### PR TITLE
Add support for docker

### PR DESCRIPTION
**Pull request by vuntz**

### CODE REVIEW

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -2457,6 +2459,11 @@ api_insecure=<%= @cinder_insecure ? 'True' : 'False' %>
 workers=<%= [node["cpu"]["total"], 2, 4].sort[1] %>
 
 
+[docker]
+# vif_driver=novadocker.virt.docker.vifs.DockerGenericVIFDriver
```
Comment (tboerger) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Does it make sense to make it configurable?

**Comment by tboerger** [comment]
:+1:

**Comment by rsalevsky** [comment]
works good +1

**Comment by mjura** [comment]
:+1:

**Comment by rossella** [comment]
+1

**Comment by jsuchome** [comment]
+1 tested, and this part of docker support works well

**Comment by vuntz** [comment]
Rebased and added some new cleanup so that we don't install libvirt on a docker node.

**Comment by jsuchome** [comment]
Tested with rebased PR, still works :-)

**Pull request merged**